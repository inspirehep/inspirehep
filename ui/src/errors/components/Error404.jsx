import React, { Component } from 'react';
import error404Image from '../images/404.svg';
import ErrorPage from './ErrorPage';

class Error404 extends Component {
  render() {
    const previousURL = document.referrer;
    return (
      <ErrorPage
        message="Sorry, we were not able to find what you were looking for..."
        imageSrc={error404Image}
        url={previousURL}
      />
    );
  }
}

export default Error404;
