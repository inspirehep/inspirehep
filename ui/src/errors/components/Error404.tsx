import React, { Component } from 'react';
import error404Image from '../images/404.svg';
import ErrorPage from './ErrorPage';

class Error404 extends Component {
  render() {
    return (
      <ErrorPage
        message="Sorry, we were not able to find what you were looking for..."
        imageSrc={error404Image}
      />
    );
  }
}

export default Error404;
