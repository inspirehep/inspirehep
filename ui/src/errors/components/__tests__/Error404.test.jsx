import React from 'react';
import { render } from '@testing-library/react';
import Error404 from '../Error404';

describe('Error404', () => {
  it('renders Error404 with correct props', () => {
    const { asFragment } = render(<Error404 />);
    expect(asFragment()).toMatchSnapshot();
  });
});
