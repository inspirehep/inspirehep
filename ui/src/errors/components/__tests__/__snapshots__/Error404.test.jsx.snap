// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 renders Error404 with correct props 1`] = `
<DocumentFragment>
  <div
    align="middle"
    class="ant-col ant-col-14 mt3 mb3"
    justify="center"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-24"
      >
        <h3
          class="f2 sm-f3"
        >
          Sorry, we were not able to find what you were looking for...
        </h3>
      </div>
    </div>
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-24"
      >
        <img
          alt="ERROR"
          src="404.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
